datus/models/openai_compatible.py

```
@@ -610,262 +662,278 @@ async def _generate_with_tools_stream_internal(
-                    # Track tool calls and results for immediate feedback
-                    temp_tool_calls = {}  # {call_id: ActionHistory}
+                agent = Agent(**agent_kwargs)
 
-                    while not result.is_complete:
-                        async for event in result.stream_events():
-                            if not hasattr(event, "type") or event.type != "run_item_stream_event":
-                                continue
+                try:
+                    result = Runner.run_streamed(agent, input=prompt, max_turns=max_turns, session=session)
+                except MaxTurnsExceeded as e:
+                    logger.error(f"Max turns exceeded in streaming: {str(e)}")
```

Ruff (0.14.14)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)